lib/http/request/writer.rb

```diff
@@ -47,7 +47,9 @@ def connect_through_proxy
       # Adds the headers to the header array for the given request body we are working
       # with
       def add_body_type_headers
-        return if @headers[Headers::CONTENT_LENGTH] || chunked?
+        return if @headers[Headers::CONTENT_LENGTH] || chunked? || (
+          @request_header[0].start_with?("GET ") && @body.source.nil?
```

Comment:
I suspect this should also apply to [HEAD](https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.2), [DELETE](https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.5), and [CONNECT](https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.6) requests, as they all have the same wording in RFC7231 about having no defined payload semantics.